File: object/src/com/object/chapter_12/Lecture.rb
class Lecture
    def initialize(name, scores)
      @name = name
      @scores = scores
    end

    def stats(this)
      "Name: #{@name}, Evaluation Method: #{this.getEvaluationMethod()}"
    end

    def getEvaluationMethod()
      "Pass or Fail"
    end
end

File: object/src/com/object/book/chapter_12/Lecture.js
function Lecture(name, scores){
    this.name = name;
    this.scores = scores;
}

Lecture.prototype.stats = function(){
    return "Name: " + this.name + ", Evaluation Method: " + this.getEvaluationMethod();
}

Lecture.prototype.getEvaluationMethod = function (){
    return "Pass or Fail"
}